### PR TITLE
Provider-Defined functions preview

### PR DESCRIPTION
**Pull request by dbanck**
Base branch for the provider-defined functions work, used as base for preview extension releases.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
+## [2.30.2024022914] (2024-02-29)
```
Comment (ansgarm) [inline CHANGELOG.md:1]: @dbanck Where is this version number come from?

Comment (ansgarm) [reply CHANGELOG.md:1]: I figured, but you did commit that number to the changelog, right? So you ran that locally and then pasted the version in there? Or did you anticipate the number based on the time you made the commit?
(I'm poking for details here, as that is lacking in the guide on creating the changelog 😄)

Comment (dbanck) [reply CHANGELOG.md:1]: 🪄

Comment (dbanck) [reply CHANGELOG.md:1]: The process of maintaining a changelog for pre-releases needs some improvement. I created the changelog commit, ran the script (which uses the commit date), copied the version, and amended the commit.

**Comment by dbanck** [comment]
Released on `main`

**Pull request closed without merging**